lib/json_api_responders/responder.rb

```diff
@@ -103,5 +111,20 @@ def resource_name
     def general_error_message
       I18n.t('json_api.errors.conflict.reason')
     end
+
+    def added_errors_obj(error_obj)
+      error_obj[:errors] = error_response unless errors
+    end
+
+    def added_resource(error_obj)
+      return unless errors
+      return error_obj[:resource] = resource.model if resource.try(:model).present?
```

Comment:
Need safe accessor &. for `resource` since it can be  nil from `respond_with_error`